volatility/framework/plugins/windows/cmdline.py

```diff
@@ -30,25 +30,39 @@ def get_requirements(cls) -> List[interfaces.configuration.RequirementInterface]
                                          optional = True)
         ]
 
+    @classmethod
+    def get_cmdline(cls, context: interfaces.context.ContextInterface, 
+                    kernel_table_name: str, proc):
+        """Extracts the cmdline from PEB
+
+        Args:
+            context: the context to operate upon
+            kernel_table_name: the name for the symbol table containing the kernel's symbols
+            proc: the process object
+
+        Returns:
+            A string with the command line
+        """
+
+        proc_id = proc.UniqueProcessId
+        proc_layer_name = proc.add_process_layer()
+
+        peb = context.object(kernel_table_name + constants.BANG + "_PEB",
+                             layer_name = proc_layer_name,
+                             offset = proc.Peb)
+        result_text = peb.ProcessParameters.CommandLine.get_string()
+
+        return result_text
+
+
     def _generator(self, procs):
 
         for proc in procs:
             process_name = utility.array_to_string(proc.ImageFileName)
-            proc_id = "Unknown"
-            try:
-                proc_id = proc.UniqueProcessId
-                proc_layer_name = proc.add_process_layer()
-            except exceptions.InvalidAddressException as excp:
-                vollog.debug("Process {}: invalid address {} in layer {}".format(proc_id, excp.invalid_address,
-                                                                                 excp.layer_name))
-                continue
+            proc_id = proc.UniqueProcessId
```

Comment:
This may end up throwing the exception we want to use for the exception message!  5:P  Better to set it to "Unknown" and then try to set it in the try/except block.

Reply:
ok, pushed mods. Sorry for the back and forth: next time I'll try to be more effective  :-)